src/types.rs

```diff
@@ -18,21 +18,28 @@ pub trait TypeValidator {
     fn validate(&self, value: &OwnedElement, issues: &mut Violations);
 }
 
+/// Provides a public facing schema type which has a reference to TypeStore
+/// to get the underlying TypeDefinition from TypeStore
```

Comment:
This is the public-facing documentation, so it shouldn't describe implementation details. This would make a good developer comment, though. (`//` vs `///`)